metaverse_modules/nameplate/model.js

```diff
@@ -0,0 +1,64 @@
+import {Box3} from 'three';
+import {Geometry} from 'three/examples/jsm/deprecated/Geometry';
+
+const addToMaterials = (materials, newMaterial) => {
+  for (let i = 0; i < materials.length; i++) {
+    const mat = materials[i];
+    if (mat.name === newMaterial.name) {
+      return [materials, i];
+    }
+  }
+  materials.push(newMaterial);
+  return [materials, materials.length - 1];
+};
+
+export const getModelGeoMat = model => {
+  const newGeometry = new Geometry();
+  let newMaterials = [];
+  model.scene.traverse(function(child) {
+    if (child.type === 'Mesh') {
+      const materialIndices = [];
+      let newItems;
+      if (child.material.length) {
+        for (let k = 0; k < child.material.length; k++) {
+          newItems = addToMaterials(newMaterials, child.material[k]);
+          newMaterials = newItems[0];
+          materialIndices.push(newItems[1]);
+        }
+      } else {
+        newItems = addToMaterials(newMaterials, child.material); // materials.push(child.material);
+        newMaterials = newItems[0];
+        materialIndices.push(newItems[1]);
+      }
+
+      if (child.geometry.isBufferGeometry) {
```

Comment:
Why can't this operate on BufferGeometry? Regardless, I don't think we need this code at all (see above).

Reply:
In case there are several children meshes in the glb model, I had to generate single geometry and matrial to implement MultiMeshInstance by merging them. So at first I tried to merge BufferGeometries into single BufferGeometry keeping children's position, but after merging their position was moved to zero point (0, 0, 0). As a result, the imported glb file wasn't similar to the original glb file and there was no way to keep their position when using BufferGeometry.
So totally, I decided to use legacy geometry to control faces and matrix to help keeping children's position. I'd like to get any hint how to keep position when using BufferGeometry. Thanks.

Reply:
The nameplate mesh should have a single model